### PR TITLE
doc(storage): improve `{Bucket,Object}Metadata` docs

### LINKED ISSUE #8429: ObjectMetadata has incorrect published documents
**Does this issue affect the google-cloud-cpp project?**
The [doxygen docs](https://googleapis.dev/cpp/google-cloud-storage/latest/classgoogle_1_1cloud_1_1storage_1_1v1_1_1ObjectMetadata.html) for GCS ObjectMetadata seem misaligned.

**What component of `google-cloud-cpp` is this related to?**
GCS

**Describe the bug**
The [doxygen docs](https://googleapis.dev/cpp/google-cloud-storage/latest/classgoogle_1_1cloud_1_1storage_1_1v1_1_1ObjectMetadata.html) for GCS ObjectMetadata seem misaligned. `has_metadata()` and `metadata()` have the same documentation as `delete_metadata(...)`

**To Reproduce**
1. Read the render docs.

**Expected behavior**
Documentation that makes sense according to the method name

**Operating system:**
N/A

**What compiler and version are you using?**
N/A

**What version of `google-cloud-cpp` are you using?**
Latest website.

**Additional context**

### PR DESCRIPTION
**Pull request by coryan**
The documentation was not rendering correctly. It was reusing the
description from one function to many unrelated functions. In addition,
the groups were also rendering incorrectly.

After re-reading the Doxygen documentation it seems that

```cc
/// @name Group description
///@{
... grouped member functions
///@}
```

Is the "Right Thing":tm:

Fixes #8429.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8434)

### CODE REVIEW

**Comment by coryan** [comment]
Rendered to PDF: 

* [object_metadata.pdf](https://github.com/googleapis/google-cloud-cpp/files/8120781/object_metadata.pdf)
* [bucket_metadata.pdf](https://github.com/googleapis/google-cloud-cpp/files/8120790/bucket_metadata.pdf)

File: google/cloud/storage/object_metadata.h [hunk 1]
```diff
@@ -88,85 +88,163 @@ inline bool operator>=(CustomerEncryption const& lhs,
...
     return *this;
   }
 
+  /// The name of the bucket containing this object.
   std::string const& bucket() const { return bucket_; }
 
+  /// The cacheControl attribute.
   std::string const& cache_control() const { return cache_control_; }
+
+  /// Set the cacheControl attributes.
```
Comment (dbolduc) [inline google/cloud/storage/object_metadata.h:124]: s/attributes/attribute/

File: google/cloud/storage/object_metadata.h [hunk 2]
```diff
@@ -88,85 +88,163 @@ inline bool operator>=(CustomerEncryption const& lhs,
...
+  /// The MD5 hash of the object contents. Can be empty.
   std::string const& md5_hash() const { return md5_hash_; }
+
+  /// The HTTPS link to access the object contents.
   std::string const& media_link() const { return media_link_; }
 
-  //@{
-  /// @name Accessors and modifiers to the `metadata` labels.
+  /**
+   * @name Accessors and modifiers for metadata metadata.
```
Comment (dbolduc) [inline google/cloud/storage/object_metadata.h:230]: metadata metadata?

File: google/cloud/storage/object_metadata.h [hunk 3]
```diff
@@ -192,56 +270,99 @@ class ObjectMetadata : private internal::CommonMetadata<ObjectMetadata> {
     return *this;
   }
 
+  /// Returns all the Object's metadata entries,
```
Comment (dbolduc) [inline google/cloud/storage/object_metadata.h:273]: s/,/./

File: google/cloud/storage/object_metadata.h [hunk 4]
```diff
@@ -192,56 +270,99 @@ class ObjectMetadata : private internal::CommonMetadata<ObjectMetadata> {
...
   std::map<std::string, std::string> const& metadata() const {
     return metadata_;
   }
+
+  /// Returns all the Object's metadata entries.
   std::map<std::string, std::string>& mutable_metadata() { return metadata_; }
-  //@}
+  ///@}
 
+  /// Returns `true` if the object has a `owner` attributes.
```
Comment (dbolduc) [inline google/cloud/storage/object_metadata.h:282]: s/a `owner`/an `owner`/

File: google/cloud/storage/object_metadata.h [hunk 5]
```diff
@@ -192,56 +270,99 @@ class ObjectMetadata : private internal::CommonMetadata<ObjectMetadata> {
...
   using CommonMetadata::has_owner;
+
+  /**
+   * The generation of the object metadata.
+   *
+   * Note that changes to the object metadata (e.g. changing the `cacheControl`
```
Comment (dbolduc) [inline google/cloud/storage/object_metadata.h:288]: optionally use `@note`

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/object_metadata.h (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/object_metadata.h]: Fixed.

File: google/cloud/storage/object_metadata.h (hunk 5 above)
Comment (coryan) [reply google/cloud/storage/object_metadata.h]: Done.

File: google/cloud/storage/object_metadata.h (hunk 4 above)
Comment (coryan) [reply google/cloud/storage/object_metadata.h]: Fixed

**Pull request merged**